tests/test_articlehistory.py

```diff
@@ -29,5 +30,76 @@ def test___init__():
     with pytest.raises(KeyError):
         raise KeyError("Data not found")
 
+
+def test_filter_by_tags():
+    '''tests filter by tags function'''
+    art = ArticleHistory("Techno", None, None, ["mobile edit"])
+    assert art.titles == "Techno"
+    assert art.user is None
+    assert art.keyword is None
+    assert art.tags == ["mobile edit"]
+    assert len(art.revisions) != 0
+
+    art2 = ArticleHistory("", None, None, [""])
+    assert art2.titles == ""
+    assert art2.user is None
+    assert art2.keyword is None
+    assert art2.tags == [""]
+    assert len(art2.revisions) == 0
+
+    art3 = ArticleHistory("Salsa (Mexican Cuisine)", None, None, [""])
+    assert art3.titles == "Salsa (Mexican Cuisine)"
+    assert art3.user is None
+    assert art3.keyword is None
+    assert art3.tags == [""]
+    assert len(art3.revisions) == 0
+
+    art4 = ArticleHistory("Cat", "Greatgiant19", None, ["mw-reverted"])
+    assert art4.titles == "Cat"
+    assert art4.user == "Greatgiant19"
+    assert art4.keyword is None
+    assert art4.tags == ["mw-reverted"]
+    assert len(art4.revisions) != 0
+
+
+def test_filter_by_keyword():
+    '''tests filter by keyword function'''
+    art = ArticleHistory("Techno", None, "techno")
+    assert art.titles == "Techno"
+    assert art.user is None
+    assert art.keyword == "techno"
+    assert art.tags is None
+    assert len(art.revisions) != 0
+
+    art2 = ArticleHistory("Techno", "Rio65trio", "techno")
+    assert art2.titles == "Techno"
+    assert art2.user == "Rio65trio"
+    assert art2.keyword == "techno"
+    assert art2.tags is None
+    assert len(art2.revisions) != 0
+
+    art3 = ArticleHistory("Cat", None, "gibberish")
+    assert art3.titles == "Cat"
+    assert art3.user is None
+    assert art3.keyword == "gibberish"
+    assert art3.tags is None
+    assert len(art3.revisions) == 0
+
+    art4 = ArticleHistory("Techno", "185.216.15.213",
+                          "techno", ["mw-reverted"])
+    assert art4.titles == "Techno"
+    assert art4.user == "185.216.15.213"
+    assert art4.keyword == "techno"
+    assert art4.tags == ['mw-reverted']
+    assert len(art4.revisions) != 0
+
+
+def test_filter():
```

Comment:
Not a big deal for now as long as we get this tested by the end of the quarter but it looks like this doesn't do anything related to the filter function.